### DISCUSSION TITLE
Email wallet by default

### DISCUSSION THREAD

**Discussion by irfanonk** [reown-com/appkit#3481] [Q&A]
When user creates a wallet with email it gives a smart account as default. We need to give email account to user because we save connected account as user wallet and force user to connect that account and also  user will export private key/mnemonic later for that specific account. 

Is there a way to disable smart account or give email account as default. 

![Screenshot 2024-12-17 at 11 58 21](https://github.com/user-attachments/assets/190ac5ae-bcae-4d93-87f8-9252a52212b3)

**Ps. Please It is urgent**

**Comment by evg-ch** [comment]
import { useAppKit, useAppKitAccount, useAppKitConnection } from '@reown/appkit/react'
import { useAccount } from 'wagmi'

const { address, addresses } = useAccount()
const { embeddedWalletInfo } = useAppKitAccount()
const { switchConnection } = useAppKitConnection({})

useEffect(() => {
if (embeddedWalletInfo) {
const isEoa = embeddedWalletInfo.accountType === 'eoa'
if (!isEoa && addresses.length > 1) {
const inactiveAddr = addresses.find((addr) => addr.toLowerCase() !== address.toLowerCase())
switchConnection({
address: inactiveAddr,
connection: { accounts: addresses.map((addr) => ({ address: String(addr) })), connectorId: 'ID_AUTH' },
})
}
}
}, [address, embeddedWalletInfo])
